source/Octopus.Client/Repositories/Async/DeploymentSettingsRepository.cs

```diff
@@ -6,68 +7,92 @@ namespace Octopus.Client.Repositories.Async
+            client = repository.Client;
         }
 
         public IDeploymentSettingsBetaRepository Beta()
         {
             return beta;
         }
+
+        public async Task<DeploymentSettingsResource> Get(ProjectResource project)
+        {
+            if (project.PersistenceSettings is VersionControlSettingsResource)
+                throw new NotSupportedException(
+                    $"Version Controlled projects are still in Beta. Use {nameof(IDeploymentSettingsBetaRepository)}.");
+
+            return await client.Get<DeploymentSettingsResource>(project.Link("Self"));
+        }
+
+        public async Task<DeploymentSettingsResource> Modify(ProjectResource project,
+            DeploymentSettingsResource deploymentSettings)
+        {
+            if (project.PersistenceSettings is VersionControlSettingsResource)
+                throw new NotSupportedException(
+                    $"Version Controlled projects are still in Beta. Use {nameof(IDeploymentSettingsBetaRepository)}.");
+
+            await client.Put(deploymentSettings.Link("Self"));
+
+            return await client.Get<DeploymentSettingsResource>(deploymentSettings.Link("Self"));
+        }
     }
 
     public interface IDeploymentSettingsBetaRepository
     {
         Task<DeploymentSettingsResource> Get(ProjectResource project, string gitref = null);
+
         Task<DeploymentSettingsResource> Modify(ProjectResource project, DeploymentSettingsResource resource,
             string commitMessage = null);
     }
 
-    class DeploymentSettingsBetaRepository : IDeploymentSettingsBetaRepository
+    internal class DeploymentSettingsBetaRepository : IDeploymentSettingsBetaRepository
     {
-        private readonly IOctopusAsyncRepository repository;
         private readonly IOctopusAsyncClient client;
+        private readonly IOctopusAsyncRepository repository;
 
         public DeploymentSettingsBetaRepository(IOctopusAsyncRepository repository)
         {
             this.repository = repository;
             client = repository.Client;
         }
 
-        public async Task<DeploymentSettingsResource> Get(ProjectResource projectResource, string gitref = null)
+        public async Task<DeploymentSettingsResource> Get(ProjectResource project, string gitref = null)
         {
-            if (!string.IsNullOrWhiteSpace(gitref))
-            {
-                var branchResource = await repository.Projects.Beta().GetVersionControlledBranch(projectResource, gitref);
+            if (!(project.PersistenceSettings is VersionControlSettingsResource settings))
+                return await repository.DeploymentSettings.Get(project);
 
-                return await client.Get<DeploymentSettingsResource>(branchResource.Link("DeploymentSettings"));
-            }
+            gitref = gitref ?? settings.DefaultBranch;
+            var branch = await repository.Projects.Beta().GetVersionControlledBranch(project, gitref);
```

Comment:
the change here to lookup the branch even when it's default was deliberate - previously it was relying on the project link, which coincidentally we've fully resolving to the default at the moment. 

We probably shouldn't do that either. At a project level we should be passing a partially resolved link.